packages/ui/src/primitives/tailwind/Tooltip/index.tsx

```diff
@@ -23,96 +23,188 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
 Infinite Reality Engine. All Rights Reserved.
 */
 
-import React, { ReactNode, useEffect, useRef, useState } from 'react'
+import React, { ReactNode, useEffect, useImperativeHandle, useRef, useState } from 'react'
+import ReactDOM from 'react-dom'
 import './tooltip.css'
 
-export interface TooltipProps {
+export interface BaseTooltipProps {
   title?: string
   content: ReactNode
   children: ReactNode
   position?: 'auto' | 'top' | 'bottom' | 'left' | 'right'
 }
 
-function Tooltip({ title, content, children, position = 'auto' }: TooltipProps) {
+export interface ControlledProps {
+  isControlled: true
```

Comment:
This is deliberately done like this, so user gets better type hints. When `isControlled` is set to `true`: Tooltip expects `onMouseEnter` and `onMouseLeave` functions as required props. Otherwise it wouldn't expect it.